.github/workflows/bench.yml

```diff
@@ -84,13 +93,15 @@ jobs:
         restore-keys: ${{ env.cache-name }}-
 
     - name: Compiled deps cache
+      id: compiled-deps
       uses: actions/cache@v2
       env:
         cache-name: compiled-deps
       with:
         path: ${{ steps.HaskEnvSetup.outputs.cabal-store }}
-        key:  ${{ env.cache-name }}-${{ runner.os }}-${{ matrix.ghc }}-${{ env.INDEX_STATE }}-${{ hashFiles('cabal.project') }}
+        key:  ${{ env.cache-name }}-${{ runner.os }}-${{ matrix.ghc }}-${{ env.INDEX_STATE }}-${{ hashFiles('cabal.project') }}-${{ hashFiles('cabal.project.freeze') }}
```

Comment:
could we not only use the freeze file instead cabal.project + freeze file?

Reply:
We use `cabal.project` & `freeze` file. Look at the line 8)

I went & looked how people do it. & do 2 caches like this is more effective organization of cache namespaces & inheritance between them.

Since `project` file changes rarely.

Reply:
I see, but all the build info which controls the cache is (or should be) in the freeze file, including the hackage index.
Also the freeze file is autogenerated, so it is stable.
If you do cosmetic changes or changes not affecting the cache at all in the cabal.project file you will invalidate the cache for no real reason.
I think the hackage index will give us a good enough base cache and all things that are in cabal.project and no in the freeze file only could throw away good caches. Otoh cabal files are not changed so frequently. And usually you need to update the hackage index to get the new dep version, so the cache is invalidated.

Reply:
Ok. Understood.

I read the https://github.com/haskell/haskell-language-server/pull/2532#discussion_r774771451 quite the other way first 8)

Reply:
Done.

Reply:
Oh. I remembered why I kept `project` - so there some symbols can be added to refresh into new keyspace. Otherwise, it is back to `index-state` changes. But after cache workflow stabilizes - it would be needed rarely.